File: Astro-TFG/public/scripts/4.1-calculos-matematicos/prueba-1/version-1/javascript/main.js
function createMatrix(size) {
    let matrix = new Array(size);
    for (let i = 0; i < size; i++) {
        matrix[i] = new Array(size);
        for (let j = 0; j < size; j++) {
            matrix[i][j] = Math.random();
        }
    }
    return matrix;
}

function multiplyMatrices(size) {
    let A = createMatrix(size);
    let B = createMatrix(size);
    let C = new Array(size).fill(0).map(() => new Array(size).fill(0));

    let start = performance.now();
    for (let i = 0; i < size; i++) {
        for (let j = 0; j < size; j++) {
            let sum = 0;
            for (let k = 0; k < size; k++) {
                sum += A[i][k] * B[k][j];
            }
            C[i][j] = sum;
        }
    }
    let end = performance.now();

    // ET (Execution Time)

    let resultTime = Number((end - start).toFixed(2));
    let result = `ET: ${resultTime} ms`;
    let resultDiv = document.createElement("div");
    resultDiv.textContent = result;
    document.getElementById("javascript-output").appendChild(resultDiv);
    document.getElementById('loading-message').style.display = 'none';
}

window.runJSBenchmark = function () {
    clearCell("javascript-output");
    window.showExecutionLoader();

    requestAnimationFrame(() => {
        setTimeout(() => {
            document.getElementById("javascript-output").textContent = ""
            multiplyMatrices(300);
            let memoryDiv = document.createElement("div");
            memoryDiv.textContent = getMemoryUsageJS();
            document.getElementById("javascript-output").appendChild(memoryDiv);
            window.hideExecutionLoader();
        }, 0);
    });

}

function getMemoryUsageJS() {
    // RAM
    if (performance.memory) {
        let memoryUsed = performance.memory.usedJSHeapSize / (1024 * 1024);
        return `RAM: ${memoryUsed.toFixed(2)} MB`;
    } else {
        return `RAM unsopported measurement in this browser.`;
    }
}

File: Astro-TFG/public/scripts/4.1-calculos-matematicos/prueba-1/version-1/python/main.py
import gc
import time
import random
import js  # type: ignore
import tracemalloc
from pyscript import display


def create_matrix(size):
    return [[random.random() for _ in range(size)] for _ in range(size)]


def multiply_matrices(size):
    tracemalloc.start()
    A = create_matrix(size)
    B = create_matrix(size)
    C = [[0] * size for _ in range(size)]

    start = time.time()
    gc.collect()
    for i in range(size):
        for j in range(size):
            _sum = 0
            for k in range(size):
                _sum += A[i][k] * B[k][j]
            C[i][j] = _sum

    execution_time = (time.time() - start) * 1000
    memory_usage = tracemalloc.get_traced_memory()[1] / (1024 * 1024)
    tracemalloc.stop()

    results = {
        "execution_time": execution_time,
        "memory_usage": memory_usage
    }

    display_results(results)
    js.endTimerWebAssembly()


def display_results(results):
    display(f"ET: {results['execution_time']:.2f} ms",
            target="pyscript-output")
    display(f"RAM: {results['memory_usage']:.2f} MB", target="pyscript-output")


def run_py_benchmark(event):
    js.clearCell('pyscript-output')
    multiply_matrices(300)


js.window.run_py_benchmark = run_py_benchmark
